### ISSUE TITLE
The strip_prompt function removes not only the prompt line but also the preceding newline.

### ISSUE THREAD

**Issue by kunitake** [ktbyers/netmiko#3526]
### Description of Issue

I think the strip_prompt function is originally intended to remove only the last line of the prompt. However, in practice, it removes not only the last prompt line but also the preceding newline.

This is generally not an issue, but when used with TextFSM, problems arise if the template uses empty lines as markers to parse data, such as with following.

>   ^$$ -> Record

When the last empty line is removed, causing the last data to not be recorded. This issue is hard to detect as it seems to be parsed correctly at first glance.

### Netmiko version

```
netmiko==4.4.0
```

### Relevant Python code

```python
class Sample1(object):
    def __init__(self, response_return, base_prompt):
        self.RESPONSE_RETURN = response_return
        self.base_prompt = base_prompt

    def strip_prompt(self, a_string: str) -> str:
        response_list = a_string.split(self.RESPONSE_RETURN)
        last_line = response_list[-1]

        if self.base_prompt in last_line:
            return self.RESPONSE_RETURN.join(response_list[:-1])
        else:
            return a_string


a_string = """aaa
bbb
ccc

[kunitake@linux ~]$"""

test_strip_prompt = Sample1('\n', '[kunitake@linux ~]$')

print('"""', end='')
print(a_string, end='"""')


print("\n" + 'strip_prompt()')
a_string = test_strip_prompt.strip_prompt(a_string)

print('"""', end='')
print(a_string, end='"""')
```

and it's result.

```
"""aaa
bbb
ccc

[kunitake@linux ~]$"""
strip_prompt()
"""aaa
bbb
ccc
"""
```

The strip_prompt function in the above program is extracted from base_connection.py.
A blank line cannot exist without a newline character.

To fix this issue, you can either add a newline character to the end of the given a_string or add a newline character to the return value of strip_prompt(). However, if there are usage examples that rely on the removal of the empty line before the prompt, this change may have side effects.

Therefore, I would appreciate it if the manual could mention the possibility of avoiding this issue by setting strip_prompt to false when using TextFSM, or if a new flag could be introduced that prevents the removal of the last empty line.

Best regards,

**Comment by ktbyers** [comment]
Probably should fix TextFSM template...the TextFSM template should not be that particular about the existence of extra white space. 

In other words, I would consider this an error/issue in the TextFSM template (caring about unimportant whitespace).

**Comment by kunitake** [comment]
You are correct.

I agree that changing the template can avoid the issue. When I encountered this problem, I also modified my template. 
However, even though it was a proven template, changing the configuration to Netmiko + TextFSM resulted in one entry missing out of hundreds of entries. This was not my expected behavior, so I thought others might have encountered a similar issue and thought it might be helpful to let you know. I hope this report is useful to someone.

I'll close this issue.

Thank you for your comment!
